### ISSUE TITLE
Test OTP verify not working on supabase local

### ISSUE THREAD

**Issue by jibin2706** [supabase/gotrue#1293]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug
Verifying OTP with Test emails is giving the following error

```
AuthApiError: Token has expired or is invalid
    at le (https://esm.sh/v133/@supabase/gotrue-js@2.56.0/esnext/gotrue-js.mjs:2:5282)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async Ie (https://esm.sh/v133/@supabase/gotrue-js@2.56.0/esnext/gotrue-js.mjs:2:6069)
    at async h (https://esm.sh/v133/@supabase/gotrue-js@2.56.0/esnext/gotrue-js.mjs:2:5806)
    at async g.verifyOtp (https://esm.sh/v133/@supabase/gotrue-js@2.56.0/esnext/gotrue-js.mjs:2:21397)
    at async Object.handler (file:///home/deno/functions/test/index.ts:14:27)
    at async handleHttp (ext:sb_core_main_js/js/http.js:66:17) {
  __isAuthError: true,
  name: "AuthApiError",
  status: 401
}
```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Run Local Instance of Supabase
2. Modify `config.toml`
```
[auth.sms]
# Allow/disallow new user signups via SMS to your project.
enable_signup = true
# If enabled, users need to confirm their phone number before signing in.
enable_confirmations = true

# Use pre-defined map of phone number to OTP for testing.
[auth.sms.test_otp]
7333333333 = "123456"
```
3. Run the following js code
```
const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY, {})
const { data, error } = await supabase.auth.verifyOtp({
		phone: '7333333333',
		token: '123456',
		type: 'sms',
	})
```

## Expected behavior

Verify the user and login

## System information

- OS: [e.g. macOS, Windows]
- Version of supabase-cli: 1.110.1
- Version of Node.js: [e.g. 18.x.x]

## Additional context

Similar to #1252

**Comment by J0** [comment]
Hey team,

Thanks for flagging this! This looks to be an error with how the `TEST_OTP` config is parsed in the CLI. We'll look into a fix and get back to you once it's out.

**Comment by LautaroJayat** [comment]
HI all!

I found something that might help the debugging of this problem.

# TL;DR
It seems the Supabase-CLI is reading  `config.toml` OK but when it creates the envs that Gotrue will use later, it mess the expected format.

Maybe changing the method used to generate the value for `GOTRUE_SMS_TEST_OTP` should be implemented.

# On the CLI side

If we see [line 421 of `start.go`](https://github.com/supabase/cli/blob/main/internal/start/start.go#L421), the value for that env being produced by calling `testOTP.String()`, which produces something like the following: `{"7333333333":"123456"}`.

Like a total "grunt" I've done what is in the screenshot bellow to check what is being produced.

![Screenshot from 2023-11-11 20-16-02](https://github.com/supabase/gotrue/assets/48366106/5827ce23-b736-48b6-82ed-84e3688cf341)

The actual output that will be used as env is shown in the screenshot bellow.

![Screenshot from 2023-11-11 20-10-13](https://github.com/supabase/gotrue/assets/48366106/e6adfb64-1409-49c1-9676-a4bbab97a5ae)

# On the Gotrue side

Now, on Gotrue, we are using the [envconfig package](https://github.com/kelseyhightower/envconfig) to parse the environments and convert them into the actual types the program expects.

If we add some logs into the [`conf.LoadGlobal` function](https://github.com/supabase/gotrue/blob/master/internal/conf/configuration.go#L384), we can check what is being produced.

![Screenshot from 2023-11-11 20-32-41](https://github.com/supabase/gotrue/assets/48366106/b3b8ce60-8afc-429a-a37b-6dd05797be70)


When using `GOTRUE_SMS_TEST_OTP={"7333333333":"123456"}` we got the following:

![Screenshot from 2023-11-11 20-43-47](https://github.com/supabase/gotrue/assets/48366106/fc625ed0-ee7e-4203-b585-828d42da93fa)


When we remove the curly brackets and the double quotes from the env value and pass `GOTRUE_SMS_TEST_OTP=7333333333:123456` we got this:

![Screenshot from 2023-11-11 20-42-42](https://github.com/supabase/gotrue/assets/48366106/6be3d0b4-6c5c-47bd-9c8f-db4228fe31d2)


FYI: @J0 & @jibin2706

**Comment by J0** [comment]
Hey @LautaroJayat,

Thanks for writing this up! That is indeed to issue I observed as well - I think the explanation is accurate- the `GOTRUE_TEST_OTP` should be a string `<number>:<otp>,<number>:<otp>` instead of the map you have illustrated above.

I haven't quite found the time to fix this but if anyone has the time to jump in please feel free. Otherwise I'll see if I can find a slot or check in with the CLI team tmrw to see if they have bandwidth.

Repo is here: https://github.com/supabase/cli

**Comment by LautaroJayat** [comment]
Hi @J0!
I will send a PR this Sunday :)
Have a nice weekend.

**Comment by jibin2706** [comment]
also is it possible to expose sms template which can we read from env?

**Comment by J0** [comment]
Thank you! Let me know if you need any assistance there and feel free to tag me for review

**Comment by LautaroJayat** [comment]
@J0 I think I saw a comment from @jibin2706 requesting a new env in the cli to configure he code template.
Should I create another PR for that?

**Comment by jibin2706** [comment]
@LautaroJayat created a PR for sms template https://github.com/supabase/cli/pull/1655

**Comment by LautaroJayat** [comment]
@J0 @jibin2706 
The fix for the env value format was addressed here https://github.com/supabase/cli/pull/1654
Shall we close this issue?

**Comment by jibin2706** [comment]
@LautaroJayat Still getting same error

The env is the following format
```
GOTRUE_SMS_TEST_OTP=917045386164:123456,917123123232:123456,917213123233:123456
```

 I guess we have to make changes to the gotrue repo as well

**Comment by J0** [comment]
@LautaroJayat Thanks so much 🙏 

@jibin2706 we shouldn't need to change the GoTrue repo I think. Can I check that you're on the latest version of the CLI?

**Comment by J0** [comment]
Hey @jibin2706 I just tested locally and it seems to work.

Can you try  changing to `config.toml` 
```
[auth.sms.test_otp]
"7333333333" = "123456"
``` 

and let me know if it works? I'll see if we need to update the config as well

**Comment by J0** [comment]
@jibin2706  do you mind sharing how you're making the verification? Here's my setup:

```
const { data, error } = await supabase.auth.signInWithOtp({
      phone: '12345678'
    })
```



```
const res = await supabase.auth.verifyOtp({
		  phone: '12345678',
		  token: '555555',
		  type: 'sms',

    })
```

and my `config.toml` contains

```
[auth.sms.test_otp]
<my_number> = "123456"
12345678 = "555555"
```

Am able to verify with this set up

**Comment by jibin2706** [comment]
working now @J0 
I was using  `auth.admin.createUser` instead `supabase.auth.signInWithOtp`

**Comment by J0** [comment]
Thanks @jibin2706! Going to close for now but let us know if there are any further issues.

**Comment by jibin2706** [comment]
@J0 
what is the release cycle of gotrue to supabase.com?

**Comment by archerabi** [comment]
For anyone still facing this issue, the sms provider check happens before the `test_otp` check. So even if you just want to use the test phone number, you have to enable an SMS provider. 

```
# Use pre-defined map of phone number to OTP for testing.
[auth.sms.test_otp]
9000001111 = "123456"

# Configure one of the supported SMS providers: `twilio`, `twilio_verify`, `messagebird`, `textlocal`, `vonage`.
[auth.sms.textlocal]
enabled = true
```
